### PR TITLE
Document the --http-addr flag.

### PR DESCRIPTION
**Pull request by glycerine**
Accompanies the merged fix https://github.com/cockroachdb/cockroach/pull/7475

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/443)

### CODE REVIEW

**Comment by glycerine** [comment]
@jseldess could you take a look?

Also CircleCI says s3 sync failed, not sure what that means. Let me know if I need to revise something.

**Comment by uptimeDBA** [comment]
I don't think the CircleCI fail is your fault @glycerine, I hit the same problem in PR #417.
From the comments: "circleci doesn't push aws credentials **for fork PRs** do avoid leaking our credentials (which makes sense). I'm not sure there's a good way around that, other than a completely public AWS bucket which isn't a great solution."
The short answer is the S3 sync part of the CircleCI process doesn't work for external contributors that are working from a fork of docs.
@jseldess is going to update the [Contributing to CockroachDB Docs](https://github.com/cockroachdb/docs/blob/gh-pages/CONTRIBUTING.md) to reflect this.

**Comment by jseldess** [comment]
Thanks for the update, @glycerine! And sorry for my delay. 

See comments below, but my overall sense is that we should demonstrate this flag not on `start-a-local-cluster.md` or `secure-a-cluster.md`, both of which focus on local clusters listening on `localhost`, but rather `manual-deployment.md`, which focuses on an insecure and secure distributed cluster. 

Paul's right about the CircleCI failure; it's out of your control. It's actually already mentioned in the docs' `contributing.md` file, but it's pretty buried. I think we'll need to find a better solution soonish.

---

Reviewed 3 of 3 files at r1.
Review status: 1 of 3 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[secure-a-cluster.md, line 49 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPsmjkSZvID9OdxnWz:-KMPsmjkSZvID9OdxnX-:-1652753081) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L49)):*
> ```Markdown
>  
>     ~~~ shell
>     $ cockroach start --ca-cert=certs/ca.cert --cert=certs/node.cert --key=certs/node.key --background --http-addr=127.0.0.1
> ```

@glycerine, this page and `start-a-local-cluster.md` assume you're running a local cluster on `localhost` only, so I don't think this is the best place to feature the `--http-addr` flag. Instead, would you mind working it into the `manual-deployment.md` page? I'm also happy to take this on, if you prefer.

---

*[secure-a-cluster.md, line 68 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPt8AQTKAI40nEfDgn:-KMPt8AQTKAI40nEfDgo:-1306023338) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L68)):*
> ```Markdown
> 
>     ~~~ shell
>     $ cockroach start --store=node2 --port=26258 --http-port=8081 --http-addr=127.0.0.1 --join=localhost:26257 --ca-cert=certs/ca.cert --cert=certs/node.cert --key=certs/node.key --background
> ```

Same as above. This page probably isn't the best place for this flag.

---

*[secure-a-cluster.md, line 69 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPtBym01MqTbOoQdc3:-KMPtBym01MqTbOoQdc4:160841994) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L69)):*
> ```Markdown
>     ~~~ shell
>     $ cockroach start --store=node2 --port=26258 --http-port=8081 --http-addr=127.0.0.1 --join=localhost:26257 --ca-cert=certs/ca.cert --cert=certs/node.cert --key=certs/node.key --background
>     $ cockroach start --store=node3 --port=26259 --http-port=8082 --http-addr=127.0.0.1 --join=localhost:26257 --ca-cert=certs/ca.cert --cert=certs/node.cert --key=certs/node.key --background
> ```

Same as above.

---

*[secure-a-cluster.md, line 122 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPtEAt60ADIiwTactF:-KMPtEAt60ADIiwTactG:691134550) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L122)):*
> ```Markdown
>     When you're done using the SQL shell, press **CTRL + D** to exit.
>  
> 7.  Reopen the [Admin UI](explore-the-admin-ui.html) by establishing an SSH tunnel `ssh -L 8080:127.0.0.1:8080 ROACHs-MBP` (substitute your first node's address for ROACHs-MBP). Then point your browser at `https://127.0.0.1:8080`. You can also find the address in the `admin` field in the standard output of any node on startup. 
> ```

Same as above. 

---

*[start-a-local-cluster.md, line 48 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPtOWO3khFkI9jQKmm:-KMPtOWO3khFkI9jQKmn:-89976994) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-local-cluster.md#L48)):*
> ```Markdown
> - Communication is insecure, with the server listening only on `localhost` on port `26257` for internal and client communication and on port `8080` for HTTP requests from the Admin UI. 
>    - To bind to different ports, set `--port=<port>` and `--http-port=<port>`. 
>    - To bind the Admin web UI to a private IP address or host, set `--http-addr=<private-addr>`.
> ```

If you're willing to demonstrate this flag's usage in `manual-deployment.md`, how about we add this, like in the bullet below: "For a demonstration, see [Manual Deployment](manual-deployment.html)."

---

*[start-a-node.md, line 37 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPwlCSNcgwGEJfbwsF:-KMPwlCTXRx8x5ci4epA:2108547851) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-node.md#L37)):*
> ```Markdown
> `--host` | The address to listen on for internal and client communication. The node also advertises itself to other nodes using this address. Therefore, if it is a hostname, it must be resolvable from all nodes, and if it is an IP address, it must be routable from all nodes.<br><br>When running an insecure local cluster (without `--insecure` and without cert flags), this defaults to `localhost` and cannot be changed. When running an insecure distributed cluster (with `--insecure` but without cert flags) or a secure local or distributed cluster (without `--insecure` but with cert flags), this can be an external address.
> `--http-port` | The port to listen on for HTTP requests from the Admin UI. <br><br>**Default:** 8080
> `--http-addr` | The IP address or hostname to listen on for HTTP requests for the Admin UI. <br><br>**Default:** same as --host
> ```

nits: change "for" to "from"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/443#-:-KMPxANiigOSOUbDj3-p:-112383644)*

**Comment by jseldess** [comment]
Review status: 1 of 3 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[start-a-node.md, line 93 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMQ-La7Y0Z5u_PX-58j:-KMQ-La7Y0Z5u_PX-58k:-101537960) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-node.md#L93)):*
> ```Markdown
> ~~~
> 
> ### Start a distributed cluster
> ```

@glycerine, should we add `http-addr` to this example as well? Since the cluster would be listening on external address, you might want to use a private address for the admin UI.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/443)*

**Comment by jseldess** [comment]
Review status: 1 of 3 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[secure-a-cluster.md, line 49 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPsmjkSZvID9OdxnWz:-KMQ2F0oTKBbVmYBPw6p:1121054709) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L49)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> @glycerine, this page and `start-a-local-cluster.md` assume you're running a local cluster on `localhost` only, so I don't think this is the best place to feature the `--http-addr` flag. Instead, would you mind working it into the `manual-deployment.md` page? I'm also happy to take this on, if you prefer.
</details>
@glycerine, my apologies. I was confused here. Since we're starting a node with certs, it listens on all interfaces by default, so it does make sense to demonstrate using `http-addr` to have the admin ui listen on a specific address. 

That said, all good here. One nit: Please move the flag before `--background`, similar to the `start` commands below.

---

*[secure-a-cluster.md, line 61 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMQ2oDvlKnad7eayjG3:-KMQ2oDwkLBYCIAkCKiQ:1734747341) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L61)):*
> ```Markdown
>     <div id="details-secure3" class="collapse" markdown="1">
> 
>     This command restarts your first node with its existing data, but securely. The command is the same as before but now uses the additional `--ca-cert`, `--cert`, and `--key` flags to point to the CA certificate and the node certificate and key created in step 2.
> ```

Could you add a sentence about the `http-addr` flag here?

---

*[secure-a-cluster.md, line 75 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMQ2vR3nJHyY5-YuHx7:-KMQ2vR3nJHyY5-YuHx8:-1282897680) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L75)):*
> ```Markdown
>     <div id="details-secure4" class="collapse" markdown="1">
> 
>     These commands restart additional nodes with their existing data, but securely. The commands are the same as before but now uses the additional `--ca-cert`, `--cert`, and `--key` flags to point to the CA certificate and the node certificate and key created in step 2.
> ```

Could you add a sentence about `http-addr` here as well?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/443)*

**Comment by glycerine** [comment]
Review status: 1 of 4 files reviewed at latest revision, 9 unresolved discussions.

---

*[secure-a-cluster.md, line 49 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPsmjkSZvID9OdxnWz:-KMQbRBkywma1cRZj3zJ:-469593337) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L49)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> @glycerine, my apologies. I was confused here. Since we're starting a node with certs, it listens on all interfaces by default, so it does make sense to demonstrate using `http-addr` to have the admin ui listen on a specific address. 
> 
> That said, all good here. One nit: Please move the flag before `--background`, similar to the `start` commands below.
</details>
done.

---

*[secure-a-cluster.md, line 61 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMQ2oDvlKnad7eayjG3:-KMQbZygaewDubKL6CzM:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L61)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Could you add a sentence about the `http-addr` flag here?
</details>
Done.

---

*[secure-a-cluster.md, line 68 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPt8AQTKAI40nEfDgn:-KMQbdAxs2CSTl9HY19N:1929739976) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L68)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Same as above. This page probably isn't the best place for this flag.
</details>
per above, we'll do both.

---

*[secure-a-cluster.md, line 69 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPtBym01MqTbOoQdc3:-KMQbhLdQSn0gOuWSJUg:1929739976) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L69)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Same as above.
</details>
per above, we'll do both.

---

*[secure-a-cluster.md, line 75 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMQ2vR3nJHyY5-YuHx7:-KMQbjfS6nC6XT3Ug-MY:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L75)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Could you add a sentence about `http-addr` here as well?
</details>
Done.

---

*[secure-a-cluster.md, line 122 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPtEAt60ADIiwTactF:-KMQblETx02URTyQ_3jZ:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/secure-a-cluster.md#L122)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Same as above. 
</details>
Done.

---

*[start-a-local-cluster.md, line 48 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPtOWO3khFkI9jQKmm:-KMQbnLOoOEo3KW0eksT:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-local-cluster.md#L48)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> If you're willing to demonstrate this flag's usage in `manual-deployment.md`, how about we add this, like in the bullet below: "For a demonstration, see [Manual Deployment](manual-deployment.html)."
</details>
Done.

---

*[start-a-node.md, line 37 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPwlCSNcgwGEJfbwsF:-KMQbokKtvOBorjVC4eK:1788113473) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-node.md#L37)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> nits: change "for" to "from"
</details>
The 'cockroach start' process is the one serving on the bound address, so that 'for' makes way more sense than 'from'. I edited the --http-port command as well to remove that same confusion.

---

*[start-a-node.md, line 93 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMQ-La7Y0Z5u_PX-58j:-KMQc5sdFB1uryogeMA8:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-node.md#L93)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> @glycerine, should we add `http-addr` to this example as well? Since the cluster would be listening on external address, you might want to use a private address for the admin UI.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/443)*

**Comment by glycerine** [comment]
Hi @jseldess,

I think I've addressed everything.

The 'for' vs. 'from' comment seemed to indicate a misunderstanding of which process was serving and which acting as client. Since we're talking about the server, I tried to clarify this language on both the port and the addr summary lines.

With that said, I'd be happy to just turn any other editing you'd like over to you to proceed as you feel best. 

- Jason

**Comment by jseldess** [comment]
:lgtm: Thanks again, Jason.

---

Review status: 1 of 4 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[start-a-node.md, line 37 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/docs/443#-KMPwlCSNcgwGEJfbwsF:-KMQtkbUAIWoHHwn9rQW:-1321490385) ([raw file](https://github.com/cockroachdb/docs/blob/2923b953bc18fc17096432ec7e35a8af028353a8/start-a-node.md#L37)):*
<details><summary><i>Previously, glycerine (Jason E. Aten, Ph.D.) wrote…</i></summary>
> The 'cockroach start' process is the one serving on the bound address, so that 'for' makes way more sense than 'from'. I edited the --http-port command as well to remove that same confusion.
</details>
Thanks, Jason. This is better. I might end up using the language in the CLI help text, but I'll figure it out in a separate pr.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/443#-:-KMQu5FDm2kW8zII6OXp:-69339777)*

**Comment by jseldess** [comment]
Reviewed 2 of 3 files at r2.
Review status: 3 of 4 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/443)*

File: manual-deployment.md [hunk 1]
```diff
@@ -142,17 +142,17 @@ Store the CA key somewhere safe and keep a backup; if you lose it, you will not
 Copy the `cockroach` binary, CA certificate, and node 1 certificate and key to the first machine and then start the node:
 
 ~~~ shell
-$ cockroach start --ca-cert=ca.cert --cert=node1.cert --key=node1.key --host=<node1-hostname>
+$ cockroach start --http-addr=127.0.0.1 --ca-cert=ca.cert --cert=node1.cert --key=node1.key --host=<node1-hostname>
```
Comment (tamird) [inline manual-deployment.md]: throughout: we should avoid "127.0.0.1" and use "localhost" instead for ipv4/ipv6 agnosticism.

File: manual-deployment.md [hunk 2]
```diff
@@ -211,17 +211,27 @@ For a list of recommended drivers that we've tested, see [Install Client Drivers
...
+admin:     https://127.0.0.1:8080 <-------------- ESTABLISH SSH TUNNEL TO HERE
 sql:       postgresql://root@node1.example.com:26257?sslcert=%2FUsers%2F...
 logs:      cockroach-data/logs
 store[0]:  path=cockroach-data
 ~~~
 
+Here is how to use ssh to tunnel port 8081 from your desktop to the node1.example.com cluster node:
+
+~~~ shell
+$ ssh -L 8081:127.0.0.1:8080 node1.example.com
```
Comment (tamird) [inline manual-deployment.md]: throughout: remove all mention of ssh tunnels - we are not in the business of teaching people how to ssh. it's enough to say that the admin UI is only available on the interface specified by `--http-addr`.

File: manual-deployment.md (hunk 1 above)
Comment (jseldess) [inline manual-deployment.md]: Addressed by https://github.com/cockroachdb/docs/pull/447

File: manual-deployment.md (hunk 2 above)
Comment (jseldess) [inline manual-deployment.md]: Addressed by https://github.com/cockroachdb/docs/pull/447

**Pull request merged**